### PR TITLE
Locale detection and language selection

### PR DESCRIPTION
**Pull request by TadeSF**
This PR implements preferred locale loading based on local storage data, as well as the ability for the unser to select the language against the browser preferred settings.

Also, the users theme preferences will now be stored as well and won't be lost if the page is reloaded.

### CODE REVIEW

File: chase/frontend/components/navbar/settings_sidebar.tsx [hunk 1]
```diff
@@ -48,21 +51,22 @@ export default function SettingsSidebar({
...
     {
       name: "Hoher Kontrast",
       icon: faCircleHalfStroke,
       value: "contrast",
     },
   ];
 
-  const [colortheme, setColortheme] = useState("system");
+  const [colortheme, setColortheme] = useState(
+    localStorage.getItem("theme") || "system",
```
Comment (TadeSF) [inline chase/frontend/components/navbar/settings_sidebar.tsx:62]: @m1212e I am lost. Can you tell me why he thinks localStorage is undefined?

```
- error components/navbar/settings_sidebar.tsx (62:4) @ localStorage
- error ReferenceError: localStorage is not defined
    at SettingsSidebar (./components/navbar/settings_sidebar.tsx:52:89)
  60 | 
  61 |   const [colortheme, setColortheme] = useState(
> 62 |     localStorage.getItem("theme") || "system",
     |    ^
  63 |   );
  64 |   const [isDarkMode, setIsDarkMode] = useState(false);
  65 | 
  ```

Comment (m1212e) [reply chase/frontend/components/navbar/settings_sidebar.tsx:62]: Try making the component client only, Browser APIs may not be available in SSR

Comment (m1212e) [reply chase/frontend/components/navbar/settings_sidebar.tsx:62]: This should do the trick

```

  const [colortheme, setColortheme] = useState(
    typeof window !== "undefined"
      ? localStorage.getItem("theme") ?? "system"
      : "system",
  );
  const [isDarkMode, setIsDarkMode] = useState(false);

  const [language, setLanguage] = useState(
    typeof window !== "undefined"
      ? localStorage.getItem("lang") ?? "system"
      : "system",
  );
```

Comment (TadeSF) [reply chase/frontend/components/navbar/settings_sidebar.tsx:62]: Thanks! That seemed so overcomplicated to me, that I thought there must be a more slick solution... I'll implement it ;)

File: chase/frontend/components/navbar/settings_sidebar.tsx [hunk 2]
```diff
@@ -26,18 +29,18 @@ interface SettingsSidebarProps {
...
- * TODO: add functionality to reset settings
  */
 
 export default function SettingsSidebar({
   settingsSidebarVisible,
   setSettingsSidebarVisible,
 }: SettingsSidebarProps) {
   const colortheme_items: ColormodeOption[] = [
+    {
+      name: "System",
```
Comment (m1212e) [inline chase/frontend/components/navbar/settings_sidebar.tsx:40]: Please add localizations

File: chase/frontend/components/navbar/settings_sidebar.tsx [hunk 3]
```diff
@@ -105,25 +111,65 @@ export default function SettingsSidebar({
...
-          onChange={(e) => {
-            setColortheme(e.value);
-          }}
-          itemTemplate={colorModeTemplate}
-          optionLabel="Color Theme"
-          options={colortheme_items}
-        />
+      <div className="flex w-full justify-center items-center pt-2 gap-2">
+        <div className="flex flex-col">
+          <div className="text-sm text-gray-500">Color Mode</div>
```
Comment (m1212e) [inline chase/frontend/components/navbar/settings_sidebar.tsx:140]: please localize

File: chase/frontend/components/navbar/settings_sidebar.tsx [hunk 4]
```diff
@@ -105,25 +111,65 @@ export default function SettingsSidebar({
...
+              setColortheme(e.value);
+            }}
+            itemTemplate={colorModeTemplate}
+            optionLabel="Color Theme"
+            options={colortheme_items}
+            unselectable={false}
+          />
+        </div>
+        <div className="flex flex-col">
+          <div className="text-sm text-gray-500">Language</div>
```
Comment (m1212e) [inline chase/frontend/components/navbar/settings_sidebar.tsx:153]: i18n

File: chase/frontend/components/navbar/settings_sidebar.tsx [hunk 5]
```diff
@@ -105,25 +111,65 @@ export default function SettingsSidebar({
...
+          <Dropdown
+            value={language}
+            options={[
+              { flag: "uno", label: "System", value: "system" },
```
Comment (m1212e) [inline chase/frontend/components/navbar/settings_sidebar.tsx:157]: i18n

File: chase/frontend/package.json [hunk 6]
```diff
@@ -37,6 +37,7 @@
     "typescript": "5.0.4"
   },
   "devDependencies": {
+    "@types/js-cookie": "^3.0.3",
```
Comment (m1212e) [inline chase/frontend/package.json:40]: what is this for?

**Review by m1212e — CHANGES_REQUESTED** [review]
Please add i18n

File: chase/frontend/package.json (hunk 6 above)
Comment (TadeSF) [reply chase/frontend/package.json:40]: A skeleton from me trying out a few options on how to store the locale over a session

**Review by m1212e — APPROVED** [review]

**Pull request merged**